controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitRequestHandler.java

```diff
@@ -306,4 +306,10 @@ public CommitRequestHandler(final StreamMetadataStore streamMetadataStore,
             return CompletableFuture.completedFuture(null);
         }
     }
+
+    @Override
+    public CompletableFuture<Boolean> hasTaskStarted(CommitEvent event) {
+        return streamMetadataStore.getState(event.getScope(), event.getStream(), true, null, executor)
+                                  .thenApply(state -> state.equals(State.COMMITTING_TXN) || state.equals(State.SEALING));
```

Comment:
Could be a problem the fact of having more than one task for which `hasTaskStarted` evaluate to `true` on a given moment? For instance, if the Stream is `SEALING` state, a commit task and a seal task would evaluate  `hasTaskStarted` to `true`. Could this ambiguity be harmful in some way?

Reply:
This is a good observation. However, this ambiguity has no harmful effect. This ambiguity is not local to fairness but it is about mutual exclusion of tasks. 
Note 1: fairness is about ensuring each processor gets a chance to execute its workflow and doesnt wait indefinitely. Fairness does not care for mutual exclusion. So if some processor is waiting to acquire a mutex to perform its task, the fairness ensures it doesnt wait long. 

Note 2: all other tasks require mutual exclusion. However, `SealStreamTask` and `commitTxn` are a special case. They are not mutually exclusive once the state is set to sealing. 
Commits should proceed even if the state is set to SEALING. 
And seal stream task doesnt start sealing the stream until all transactions have been completed (either committed or aborted). 
So once the state is set to sealing, both tasks can run concurrently and dont require mutual exclusion with each other. 

When either task is picked for processing, and if the state is SEALING, it determines that it can be performed and does not request for a "waiting processor" for fairness or ignores one if present because the processing for this task has already been initiated and should be proceeded with until it is completed.